modules/core/04-channel/types/upgrade.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
+
+	"github.com/cosmos/ibc-go/v7/internal/collections"
+)
+
+// ValidateBasic performs a basic validation of the upgrade fields
+func (u Upgrade) ValidateBasic() error {
+	if err := u.Fields.ValidateBasic(); err != nil {
+		return errorsmod.Wrap(err, "proposed upgrade fields are invalid")
+	}
+
+	if !u.Timeout.IsValid() {
+		return errorsmod.Wrap(ErrInvalidUpgrade, "upgrade timeout cannot be empty")
+	}
+
+	// TODO: determine if last packet sequence sent can be 0?
+	return nil
+}
+
+// ValidateBasic performs a basic validation of the proposed upgrade fields
+func (uf UpgradeFields) ValidateBasic() error {
+	if !collections.Contains(uf.Ordering, []Order{ORDERED, UNORDERED}) {
```

Comment:
I wonder if there is a variable we can use for `[]Order{ORDERED, UNORDERED}`? Such as `SupportedOrderings`?